test-tests/shared/src/test/scala/zio/test/environment/EnvironmentSpec.scala

```diff
@@ -0,0 +1,62 @@
+package zio.test.environment
+
+import java.util.concurrent.TimeUnit
+
+import zio._
+import zio.duration._
+import zio.test.Assertion._
+import zio.test._
+
+object EnvironmentSpec
+    extends ZIOBaseSpec(
+      suite("EnvironmentSpec")(
+        testM("Clock returns time when it is set") {
+          for {
+            _    <- TestClock.setTime(1.millis)
+            time <- clock.currentTime(TimeUnit.MILLISECONDS)
+          } yield assert(time, equalTo(1L))
+        },
+        testM("Console writes line to output") {
+          for {
+            _      <- console.putStrLn("First line")
+            _      <- console.putStrLn("Second line")
+            output <- TestConsole.output
+          } yield assert(output, equalTo(Vector("First line\n", "Second line\n")))
+        },
+        testM("Console reads line from input") {
+          for {
+            _      <- TestConsole.feedLines("Input 1", "Input 2")
+            input1 <- console.getStrLn
+            input2 <- console.getStrLn
+          } yield {
+            assert(input1, equalTo("Input 1"))
```

Comment:
You need an `&&` here. Since assertions are just values a bare statement has no effect.

Reply:
i still have my imperative hat on :(